### PR TITLE
feat: spin render when deleting file

### PR DESCRIPTION
**Pull request by Samuel-Brito19**
Adding spinner to indicate when the user delete a file.

Issue: #62

### CODE REVIEW

**Comment by vieiralucas** [comment]
Hi @Samuel-Brito19 can you sign the CLA?

**Comment by Samuel-Brito19** [comment]
Done!

File: apps/web/src/components/Files.tsx [hunk 1]
```diff
@@ -424,8 +424,12 @@ function FileItem(props: FileItemProps) {
...
   }, [props.onUseInSQL, props.file])
 
-  const onRemove = useCallback(() => {
-    props.onDelete(props.file)
+  const [isDeleting, setIsDeleting] = useState(false)
+
+  const onRemove = useCallback(async () => {
+    setIsDeleting(true)
+    await props.onDelete(props.file)
+    setIsDeleting(false)
```
Comment (vieiralucas) [inline apps/web/src/components/Files.tsx:432]: I think we should use the existing `isDeleting` prop.
What we need to do here is debug why it is not becoming `true` when the data source is being removed.

We should check the parent of this component, check what it is giving as the `isDeleting` prop and why it is not passing `true` while the file is being deleted.

**Review by vieiralucas — CHANGES_REQUESTED** [review]
Hello @Samuel-Brito19 

Thank you so much for your PR, what you did here works but I would like to do it differently, can you check my comment and let me know if it makes sense?

Thank you.

File: apps/web/src/components/Files.tsx [hunk 2]
```diff
@@ -309,7 +309,7 @@ file`
                           onUseInPython={onUseInPython}
                           onUseInSQL={onUseInSQL}
                           onDelete={onRemove}
-                          isDeleting={deleting[file.path]}
+                          isDeleting={deleting[file.name]}
```
Comment (vieiralucas) [inline apps/web/src/components/Files.tsx:312]: genius!

**Pull request merged**